asapdiscovery-docking/asapdiscovery/docking/scripts/single_target_docking.py

```diff
@@ -0,0 +1,416 @@
+        if not args.title:
+            logger.info(
+                "No title provided, MD5 hashing SMILES string to generate title, consider providing a title with --title"
+            )
+            args.title = (
+                "TARGET_MOL-" + hashlib.md5(args.mols.encode("utf-8")).hexdigest()
+            )
+
+        exp_data = [ExperimentalCompoundData(compound_id=args.title, smiles=args.mols)]
+    else:
+        mol_file_path = Path(args.mols)
+        if mol_file_path.suffix == ".smi":
+            logger.info(f"Input molecules is a SMILES file: {args.mols}")
+            exp_data = oe_load_exp_from_file(args.mols, "smi")
+        elif mol_file_path.suffix == ".sdf":
+            logger.info(f"Input molecules is a SDF file: {args.mols}")
+            exp_data = oe_load_exp_from_file(args.mols, "sdf")
+        else:
+            raise ValueError(
+                f"Input molecules must be a SMILES file, SDF file, or SMILES string. Got {args.mols}"
+            )
+
+    logger.info(f"Loaded {len(exp_data)} molecules.")
+
+    if args.verbose:
+        for exp in exp_data:
+            logger.info(f"Loaded molecule {exp.compound_id}: {exp.smiles}")
+
+    # parse prep arguments
+    prep_dir = output_dir / "prep"
+    prep_dir.mkdir(parents=True, exist_ok=True)
+    intermediate_files.append(prep_dir)
+
+    logger.info(f"Prepping receptor in {prep_dir} at {datetime.now().isoformat()}")
+
+    # check inputs
+    receptor = Path(args.receptor)
+    if receptor.suffix != ".pdb":
+        raise ValueError(f"Receptor must be a PDB file. Got {args.receptor}")
+
+    if not receptor.exists():
+        raise ValueError(f"Receptor file does not exist: {args.receptor}")
+
+    if args.ref_prot is not None:
+        ref_prot = Path(args.ref_prot)
+        if ref_prot.suffix != ".pdb":
+            raise ValueError(
+                f"Reference protein must be a PDB file. Got {args.ref_prot}"
+            )
+
+    if args.loop_db is not None:
+        loop_db = Path(args.loop_db)
+        if not loop_db.exists():
+            raise ValueError(f"Loop database file does not exist: {args.loop_db}")
+
+    if args.seqres_yaml is not None:
+        seqres_yaml = Path(args.seqres_yaml)
+        # check it exists
+        if not seqres_yaml.exists():
+            raise ValueError(f"SEQRES yaml file does not exist: {args.seqres_yaml}")
+
+        # load it
+        logger.info(f"Using SEQRES from {args.seqres_yaml}")
+        with open(args.seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+    else:
+        seqres = None
+
+    # load the receptor
+    receptor_name = receptor.stem
+    xtal = CrystalCompoundData(
+        str_fn=args.receptor, smiles=None, output_name=str(receptor_name)
+    )
+
+    logger.info(f"Loaded receptor {receptor_name} from {receptor}")
+    oe_prep_function(
```

Comment:
this function is likely to be changed as we consolidate the protein prepping scripts in #251

Reply:
roger!